### ISSUE TITLE
EXTERNAL for sasl-config

### ISSUE THREAD

**Issue by visibletrap** [michaelklishin/langohr#68]
I'm the one who made this PR https://github.com/ruby-amqp/rubybunny.info/pull/27, but when it comes to langohr, the story is different.

We had an error when trying to make a rabbbitmq connection to rabbitmq broker that has [ssl authentication plugin](http://github.com/rabbitmq/rabbitmq-auth-mechanism-ssl) enabled. We figured through looking at Java client source code that we have to set `DefaultSaslConfig/EXTERNAL` to `sasl-config` to make it works, as in the snippet below.
```Clojure
(rmq/connect { ...
               :sasl-config DefaultSaslConfig/External }
```

I need your thought and help on these.
* What's an appropriate place to put this example in the [langohr.docs](https://github.com/clojurewerkz/langohr.docs)?
* Should we wrap `DefaultSaslConfig/External` in a more friendly fashion? Like for example
```Clojure
{:sasl-config "EXTERNAL"}
```
I can try to come up with a change PR if you think it's a good way to go.

**Comment by michaelklishin** [comment]
@visibletrap yes, a new option (`:authentication-mechanism`) would be great. The right place to cover this is next to other TLS options.

Thank you!

**Comment by visibletrap** [comment]
Sorry for being slow on implementing this. Finally I get a chance to dig into it. I have enough implementation to make this work here https://github.com/michaelklishin/langohr/commit/a1067ed78af37acba57d8d6cf609d685add3bb77. But I'm struggling and need some advices whether do I need to write test for this? 

On one side, I think it's good to have test as a documentation as [you mention here](https://github.com/michaelklishin/langohr#code-examples). But on the other side, test for this is required a bit of setup. It needs a client certificate that have CN matches with user name in rabbitmq server. I'm not sure the amount of complexity introduced with the test is worth covering the tiny bit of code.

Also I do fix/improve the existing tls-test here https://github.com/michaelklishin/langohr/compare/ee13462...22be9aa, would you interest in merging those in?

**Comment by michaelklishin** [comment]
@visibletrap different authentication mechanisms are a pain to test, so given how trivial this change is, I'm OK with not having tests for it, given that you test it adequately manually.

**Comment by visibletrap** [comment]
@michaelklishin I have manually tested this (with / without the option) extensively.

**Comment by michaelklishin** [comment]
Should be fixed by #70. Thanks again, @visibletrap :)

**Comment by visibletrap** [comment]
@michaelklishin Great. Thanks for you help and super fast response!
